scripts/dashboard-esbuild.ts

```diff
@@ -0,0 +1,427 @@
+    mainScssPath
+  ]
+
+  try {
+    const result = await esbuild.build({
+      entryPoints,
+      bundle: true,
+      outdir: outDir,
+      format: 'esm',
+      target: 'es2020',
+      platform: 'browser',
+      splitting: true,
+      chunkNames: 'assets/js/[name]-[hash]',
+      entryNames: 'assets/js/[name]',
+      assetNames: 'assets/fonts/[name]',
+      sourcemap: false,
+      minify: false,
+      define: {
+        'process.env.NODE_ENV': '"development"'
+      },
+      alias: {
+        '@common': './src/serve/dashboard/common',
+        '@model': './src/serve/dashboard/model',
+        '@controller': './src/serve/dashboard/controller',
+        '@view-utils': './src/serve/dashboard/views/utils',
+        '@views': './src/serve/dashboard/views',
+        '@components': './src/serve/dashboard/views/components',
+        '@containers': './src/serve/dashboard/views/containers',
+        '@pages': './src/serve/dashboard/views/pages',
+        '@forms': './src/serve/dashboard/views/components/forms',
+        '@validators': './src/serve/dashboard/views/utils/validators',
+        '@assets': './src/serve/dashboard/assets',
+        'node:process': 'process'
+      },
+      loader: {
+        '.vue': 'js', // Will be handled by our Vue plugin
+        '.png': 'file',
+        '.jpg': 'file',
+        '.jpeg': 'file',
+        '.gif': 'file',
+        '.svg': 'file',
+        '.woff': 'file',
+        '.woff2': 'file',
+        '.ttf': 'file',
+        '.eot': 'file'
+      },
+      external: [
+
+      ],
+      plugins: [
+        sassPlugin({
+          type: 'css',
+        }),
+        vuePlugin({ aliases: {} }),
+        // Plugin to resolve npm: prefixes (as suggested by @corrideat)
+        {
+          name: 'npm-prefix-resolver',
+          setup (build) {
+            build.onResolve({ filter: /^npm:/ }, args => {
+              if (args.namespace !== 'file') return
+              const { path, ...extra } = args
+              return build.resolve(path.slice(4), extra)
+            })
+          }
+        } as esbuild.Plugin,
+        // Process polyfill plugin
+        {
+          name: 'process-polyfill',
+          setup (build) {
+            build.onResolve({ filter: /^node:process$/ }, () => ({
+              path: 'process-polyfill',
+              namespace: 'process-polyfill'
+            }))
+
+            build.onLoad({ filter: /.*/, namespace: 'process-polyfill' }, () => ({
+              contents: `
+                // Process polyfill for browser environment
+                const process = {
+                  env: typeof globalThis !== 'undefined' && globalThis.process ? globalThis.process.env : {},
+                  cwd: () => '/',
+                  platform: 'browser',
+                  version: '16.0.0',
+                  versions: { node: '16.0.0' }
+                };
+                module.exports = process;
+              `,
+              loader: 'js'
+            }))
```

Comment:
Why is this here?

The only reason we use `process` on the frontend AFAIK is for stuff like `process.env.NODE_ENV`, which is supposed to be replaced by string literals during the compilation step.

So this polyfill should be removed.